src/main/java/sirius/kernel/nls/NLS.java

```diff
@@ -1136,14 +1138,14 @@ public static <V> V parseUserString(Class<V> clazz, String value, String languag
     private static <V> V parseBasicTypesFromUserString(Class<V> clazz, String value, String language) {
         if (Integer.class.equals(clazz) || int.class.equals(clazz)) {
             try {
-                return (V) Integer.valueOf(value);
+                return (V) extractIntegerFromDouble(() -> parseDecimalNumberFromUser(value, language));
```

Comment:
can't we make parseDecimalNumberFromUser using BigDecimal or Amount and then use longValueExact and intValueExact - this would give use the best of all worlds... and it would add proper support for numbers > 2^56